CS157A-Team18/CS157A-Team18
# [D-47] Migrate Authentication to Firebase

Changes:
- Install firebase dependency
- Migrate authentication to Firebase
- Remove old login endpoint
- Clean up `handleSubmit()` function

To Test:
1. Pull this branch
2. In the `client` directory, do `npm install`
3. Ensure the `.env` file in the `client` folder is updated with the one in the `React .env` folder on the Google Drive
4. Do `npm start`
5. Open another terminal and navigate to the root directory. Do `npm start`
6. Create a random user in the signup page. **NOTE: The username must be an email address.** In the console, you should see something like this:
![image](https://user-images.githubusercontent.com/36905285/69771408-8af3c780-1141-11ea-8d11-36524ebc706f.png)
7. Go to the Firebase console and look for the UID of the user you just created. Open MySQL workbench and do the following query: `SELECT * FROM user WHERE uid = '<uid goes here>'`. You should see the something like this:
![image](https://user-images.githubusercontent.com/36905285/69771600-40267f80-1142-11ea-87bd-ba934991567e.png)
8. Test logging into the user you just created. You should see the following result if the user is logged in successfully:
![image](https://user-images.githubusercontent.com/36905285/69771652-706e1e00-1142-11ea-89d8-e13c9c676dac.png)

The following result will show if the user was not logged in successfully because of a wrong password:
![image](https://user-images.githubusercontent.com/36905285/69771676-84b21b00-1142-11ea-833e-a1386fbecece.png)
